pkg/scheduler/core/equivalence/eqivalence.go

```diff
@@ -282,6 +291,7 @@ func (n *NodeCache) lookupResult(
 	n.mu.RLock()
 	defer n.mu.RUnlock()
 	value, ok = n.cache[predicateKey][equivalenceHash]
+	delete(n.receivedInvalidationRequests, predicateKey)
```

Comment:
`lookupResult` is in the critical path during scheduling, but this delete operation requires `mu.Lock()` instead of `mu.RLock()`.
 
We may need evaluation of performance impact here.

Reply:
With new solution misterikkit suggested, `RW` lock is not required now.